### ISSUE TITLE
Passing spreading array to function parameters should raise an error

### ISSUE THREAD

**Issue by zanminkian** [microsoft/TypeScript#59390]
### 🔎 Search Terms

spreading array function parameter

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about spreading array in function parameters.

### 💻 Code

```ts
function foo(args?: string): unknown {
  return args;
}
declare const arr: string[];

foo(...arr);
```

### 🙁 Actual behavior

No compilation error.

### 🙂 Expected behavior

TS compilation error when calling `foo(...arr)`

## One
```ts
function foo(): unknown
function foo(args: string): unknown
function foo(args?: string): unknown {
  return args;
}
declare const arr: string[];

foo(...arr); // ts compilation error here as expected 🤔️
```

## Two
Background

```ts
function getNames() {
  return ['foo', 'bar', 'foo'];
}

const uniqueNames = [...new Set(...getNames())]; // I mistakenly spreading the `getNames()`
console.log(uniqueNames) // [ 'f', 'o' ]
```
Once I've wrote code above. What I expect is `[ 'foo', 'bar' ]`. It's really hard to find out the problem in long-line codes, if TS not reports it in compilation time.

**Comment by MartinJohns** [comment]
This is working as intended. See https://github.com/microsoft/TypeScript/issues/48575#issuecomment-1092189084:

> The behavior we actually work hard to exhibit here is that a spread is valid if all its possible arities result in valid calls, [...]

I think your additional "One" does not work because overloads are not supported for this case.  
And the "Two" version works unfortunately due to #52060 / #9726.
